state-chain/custom-rpc/src/lib.rs

```diff
@@ -45,6 +46,38 @@ use std::{
 	sync::Arc,
 };
 
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct ScheduledSwap {
```

Comment:
Rather than repeating this, couldn't this type be defined as:

```
pub struct ScheduledSwap {
	#[serde(flatten)]
	swap_leg: SwapLegInfo,
	execute_at: BlockNumber,
}
```

(and then derive de/serialize traits on SwapLegInfo with appropriate annotations).

Do we do it this way just so we can use NumberOrHex? 


*edit: it looks like this how it was done previously* 😅

Reply:
Can we use serde_with?

```rust
#[serde(with = "hex::serde")]
amount: AssetAmount
```

If we're happy to use NumberOrHex we also have a utility:
```
#[serde(with = "utilities::serde::number_or_hex")]
```

Reply:
We could create a `#[serde(with = "utilities::serde::u256")]`... If you feel like it. Otherwise I don't mind duplicating with explicit conversion to U256.

Reply:
I experimented with adding `utilities::serde::u256`, but looks like it doesn't compose nicely with `Option`. I'm sure it is possible to make it work, but didn't seem worth spending more time on it, so I will keep the separate struct for now.